src/common/color/convert-color.ts

```diff
@@ -1,162 +1,163 @@
-import colors from "color-name";
-import { expandHex } from "./hex";
-
-const rgb_hex = (component: number): string => {
-  const hex = Math.round(Math.min(Math.max(component, 0), 255)).toString(16);
-  return hex.length === 1 ? `0${hex}` : hex;
-};
-
-// Conversion between HEX and RGB
-
+import {
+  convertHsvToRgb,
+  convertLabToRgb,
+  convertRgbToHsv,
+  convertRgbToLab,
+  formatHex,
+  parse,
+} from "culori";
+
+/**
+ * Converts a hex color string to an RGB array.
+ * @param hex - The hex color string to convert.
+ * @returns The RGB array.
+ * @throws If the hex color is invalid.
+ */
 export const hex2rgb = (hex: string): [number, number, number] => {
-  hex = expandHex(hex);
-
+  const color = parse(hex);
+  if (!color || color.mode !== "rgb") {
+    throw new Error(`Invalid hex color: ${hex}`);
+  }
```

Comment:
Why not use `converter('rgb')`, so it works with everything and we don't have to throw